xdsl/dialects/func.py

```diff
@@ -43,6 +43,19 @@ def from_callable(name: str, input_types: List[Attribute],
                           ])
         return op
 
+    @staticmethod
+    def external(name: str, input_types: List[Attribute],
```

Comment:
I am fine with adding this, just for my understanding: This is the same as calling `from_region` with an empty region, right? Or is there another subtle difference I missed? How is this implemented in MLIR, I am wondering?

Reply:
Yep, this is basically the same.

Reply:
Yep, I have no clue why external functions do not have optional region.
Or rather I think I know why. It's probably because before there weren't any notions of optional regions ;)